pallets/identity/src/lib.rs

```diff
@@ -1671,22 +1671,21 @@ impl<T: Trait> Module<T> {
     /// # Errors
     /// - `CDDIdNotUniqueForIdentity` is returned when new cdd claim's cdd_id
     /// doesn't match the existing cdd claim's cdd_id.
-    fn ensure_cdd_id_uniqueness(claim: &Claim, target: IdentityId) -> DispatchResult {
+    fn ensure_cdd_id_uniqueness(
+        claim: &Claim,
+        issuer: IdentityId,
+        target: IdentityId,
+    ) -> DispatchResult {
         if let Claim::CustomerDueDiligence(cdd_id) = claim {
             ensure!(
-                !Self::base_fetch_valid_cdd_claims(target, 0.into(), None).any(|id_claim| {
-                    if let Claim::CustomerDueDiligence(c_id) = id_claim.claim {
-                        // Allowing default cdd_id.
-                        if !c_id.is_default_cdd()
-                            && !SYSTEMATIC_ISSUERS
-                                .iter()
-                                .any(|si| si.as_id() == id_claim.claim_issuer)
-                        {
-                            return c_id != *cdd_id;
+                (cdd_id.is_default_cdd()
+                    && SYSTEMATIC_ISSUERS.iter().any(|si| si.as_id() == issuer))
+                    || Self::base_fetch_valid_cdd_claims(target, 0.into(), None).all(|id_claim| {
```

Comment:
There should not be any || case when `cdd_id.is_default_cdd()` is true but `SYSTEMATIC_ISSUERS.iter().any(|si| si.as_id() == issuer)`. 

Add a different ensure first - `!cdd_id.is_default_cdd() || SYSTEMATIC_ISSUERS.iter().any(|si| si.as_id() == issuer)` and remove the issuer check from the next ensure.

Reply:
> There should not be any || case when cdd_id.is_default_cdd() is true but SYSTEMATIC_ISSUERS.iter().any(|si| si.as_id() == issuer) is not.

What do you mean by that? There can be a case where CDD provider can pass the default cdd_id to avoid that we have that check where if cdd_id is default then the issuer should also be systematic issuers. 

and we have `||` to check the condition when cdd_id is not default.

Reply:
In the current logic, it will return success even when cdd_id is `default` and the CDD provider is NOT a systematic CDD provider as long as there isn't any other c_id attached to the user.

See https://github.com/PolymathNetwork/Polymesh/commit/e03c10d7834ee4d7df38c53da9992613e2d86710 for fix and how it breaks dozens of test cases that depend on creating default cdd_id from non-systematic cdd providers.